Code/Robot/Robot/Robot.ino

```diff
@@ -135,136 +217,219 @@ void read_serial(JsonDocument& doc) {
-  int delayTime = nextMove.time;
-  int motorMax = 400;
-  switch ((Directions)nextMove.direction) {
-    DEBUG_PRINT("NEXTMOVE.DIRECTION: ");
-    DEBUG_PRINTLN(nextMove.direction);
+  return retMove;
+}
 
-    case eForwards:  // forwards
-      DEBUG_PRINTLN("Driving: Forwards");
+void executeFreeDrive(Move nextMove) {
+  float wheelSpeeds[cNumberOfWheels];  // Initialize motor speeds
+  int delayTime = nextMove.params.freedriveParams.duration;
+
+  switch ((Directions)nextMove.params.freedriveParams.direction) {
+    case eForwards:
       gWheelbase->computeWheelSpeeds(0, 10, 0, wheelSpeeds);
-      runMotorsWithBlockingDelay(delayTime, wheelSpeeds, motorMax, false);
+      runMotorsWithBlockingDelay(delayTime, wheelSpeeds);
       break;
-    case eLeft:  // left
-      DEBUG_PRINTLN("Driving: Left");
+    case eLeft:
       gWheelbase->computeWheelSpeeds(-10, 0, 0, wheelSpeeds);
-      runMotorsWithBlockingDelay(delayTime, wheelSpeeds, motorMax, false);
+      runMotorsWithBlockingDelay(delayTime, wheelSpeeds);
       break;
-    case eBackwards:  // backwards
-      DEBUG_PRINTLN("Driving: Backwards");
+    case eBackwards:
       gWheelbase->computeWheelSpeeds(0, -10, 0, wheelSpeeds);
-      runMotorsWithBlockingDelay(delayTime, wheelSpeeds, motorMax, false);
+      runMotorsWithBlockingDelay(delayTime, wheelSpeeds);
       break;
-    case eRight:  // right
-      DEBUG_PRINTLN("Driving: Right");
+    case eRight:
       gWheelbase->computeWheelSpeeds(10, 0, 0, wheelSpeeds);
-      runMotorsWithBlockingDelay(delayTime, wheelSpeeds, motorMax, false);
+      runMotorsWithBlockingDelay(delayTime, wheelSpeeds);
       break;
-    case eCCW:  // rotate ccw
-      DEBUG_PRINTLN("Driving: rotate ccw");
-      // Rotations are more sensitive. I hand calculated 1.059 to match our map scale
+    case eCCW:
       gWheelbase->computeWheelSpeeds(0, 0, 1.059, wheelSpeeds);
-      runMotorsWithBlockingDelay(delayTime, wheelSpeeds, motorMax, false);
+      runMotorsWithBlockingDelay(delayTime, wheelSpeeds);
       break;
-    case eCW:  // rotate cw
-      DEBUG_PRINTLN("Driving: rotate cw");
-      // Rotations are more sensitive. I hand calculated 1.059 to match our map scale
+    case eCW:
       gWheelbase->computeWheelSpeeds(0, 0, -1.059, wheelSpeeds);
-      runMotorsWithBlockingDelay(delayTime, wheelSpeeds, motorMax, false);
-      break;
-    case eLift:  // lift motor
-      DEBUG_PRINTLN("Driving: lift motor");
-      runMotorsWithBlockingDelay(delayTime, nullptr, motorMax, true);
-      break;
-    case eBelt:  // belt motor
-      DEBUG_PRINTLN("Driving: belt motor");
-      runMotorsWithBlockingDelay(delayTime, nullptr, motorMax, false);
+      runMotorsWithBlockingDelay(delayTime, wheelSpeeds);
       break;
     default:
-      DEBUG_PRINT("Unexpected input in direction switch: ");
-      DEBUG_PRINTLN(nextMove.direction);
+      DEBUG_PRINTLN("Unexpected input in direction switch for freedrive.");
       break;
   }
```

Comment:
```suggestion
  switch (nextMove.params.freedriveParams.direction) {
    case eForwards:
      gWheelbase->computeWheelSpeeds(0, 10, 0, wheelSpeeds);
      runMotorsWithBlockingDelay(delayTime, wheelSpeeds);
      break;
    case eLeft:
      gWheelbase->computeWheelSpeeds(-10, 0, 0, wheelSpeeds);
      runMotorsWithBlockingDelay(delayTime, wheelSpeeds);
      break;
    case eBackwards:
      gWheelbase->computeWheelSpeeds(0, -10, 0, wheelSpeeds);
      runMotorsWithBlockingDelay(delayTime, wheelSpeeds);
      break;
    case eRight:
      gWheelbase->computeWheelSpeeds(10, 0, 0, wheelSpeeds);
      runMotorsWithBlockingDelay(delayTime, wheelSpeeds);
      break;
    case eCCW:
      gWheelbase->computeWheelSpeeds(0, 0, 1.059, wheelSpeeds);
      runMotorsWithBlockingDelay(delayTime, wheelSpeeds);
      break;
    case eCW:
      gWheelbase->computeWheelSpeeds(0, 0, -1.059, wheelSpeeds);
      runMotorsWithBlockingDelay(delayTime, wheelSpeeds);
      break;
    default:
      DEBUG_PRINTLN("Unexpected input in direction switch for freedrive.");
      break;
  }
```

This cast will become unnecessary if we choose to store the Direction enum in the freeMoveParams union struct.